src/v2/view-builder/views/TerminalView.js

```diff
@@ -124,16 +135,18 @@ const Body = BaseForm.extend({
 });
 
 const Footer = BaseFooter.extend({
+  // All terminal views should have Back to sign in link either widget configured or server configured.
   links: function() {
-    if (this.options.appState.containsMessageWithI18nKey(GET_BACK_TO_SIGN_LINK_FLOWS)) {
-      return getBackToSignInLink(this.options.settings);
-    }
     if (this.options.appState.containsMessageStartingWithI18nKey(SAFE_MODE_KEY_PREFIX)) {
       return getSkipSetupLink(this.options.appState);
     }
     if (this.options.appState.containsMessageWithI18nKey(DEVICE_CODE_ERROR_KEYS)) {
       return getReloadPageButtonLink();
     }
+
+    if (!this.options.appState.containsMessageWithI18nKey(NO_BACKTOSIGNIN_LINK_VIEWS)) {
+      return getBackToSignInLink(this.options.settings);
+    }
```

Comment:
Double checking the consequences of this change - are we expecting the safe-mode and device code links to take precedence over the "Back to sign in" link?

Reply:
Good point, though only one error will occur and only one link will be shown.. for eg.. for safe-mode it should be `skip setup link` if i raise `back to sign in` link to the top then `skip setup link` link won't be shown..

Reply:
Yep - we should probably sync with UX since this is changing. I believe this is the behavior we want, but we should confirm.